SoColissimo.php

```diff
@@ -69,7 +69,7 @@ public static function getPrices()
      */
     public function isValidDelivery(Country $country)
     {
-        $cartWeight = $this->getRequest()->getSession()->getCart()->getWeight();
+        $cartWeight = $this->getRequest()->getSession()->getSessionCart()->getWeight();
```

Comment:
The ```getSessionCart``` method sometimes require the event_dispatcher.

As you have the container in this context, you can do: ```this->getRequest()->getSession()->getSessionCart($this->container->get("event_dispatcher"))->getWeight();```